### ISSUE TITLE
Mixing ModelList and LinkLists

### ISSUE THREAD

**Issue by andybak** [django-admin-tools/django-admin-tools#29]
I like my dashboard panels to be grouped by theme.

Therefore - it is useful to be able to intersperse custom admin pages (via links) with changelists:

        self.children.append(MixedList(
            _('Newsletter'),
            children=(
                {models: 'newsletter.models.*'},
                {'title': 'Image library', 'url': reverse('filebrowser:fb_browse')},
            )
        ))
 
This isn't hard to do but does involve duplicating a fair amount of code from ModelList and LinkList. If I refactored these two classes I could eliminate this. Would this be a contribution worth considering for the release? If so - I'll do it 'properly'. Backwards compatibility shouldn't be a huge issue unless someone is doing something fairly strange with either of these two classes.

**Comment by izimobil** [comment]
@andybak OK for me but I suggest we merge the changes after the new release. I'd like to release the new version in the next days.

**Comment by andybak** [comment]
Just thinking about this. Some options:

1. We make ModelList itself more flexible - if a child item has title/url instead of a 'models' key then it does the right thing.

2. We replace LinkList and ModelList with a new class that is smart enough to handle mixed content and keep the old names around as deprecated aliases. 

3. Keep LinkList and ModelList exactly as they are and create a new class. 

Currently I think I prefer the options in this order with 1 being my favourite.

2 and 3 require a new name for the class. Not sure I like 'MixedList' very much. Any thoughts? And any thoughts on other functionality that it might be worth adding at the same time as this change?

**Comment by izimobil** [comment]
I'm not very happy with the current API, I think that in the long term we should completely rethink the way dashboards and menus are built, and make it much simpler with shortcuts that are smart enough to decide what module should be used.

In the meantime, I'd vote for your solution 3: "MixedList", just because it's the safest solution and it will not introduce regressions.

**Comment by andybak** [comment]
Got any examples of the kind of thing you're thinking of? It might be without much extra effort this could be a small step towards it.

**Comment by izimobil** [comment]
No, I'm not sure of the best way to simplify the API, we need to think about it. Maybe something like:

```python
class MyDashboard(dashboards.Dashboard):
    def init_with_context(self, context):
        self.add_module("Some title", [
            {
                'model_list': ['app.models.*', 'app2.models.SomeModel'],
                'exclude': ['app.models.ExcludedModel'],
            },
            {
                'links': [{'title': 'Django', url: 'https://www.djangoproject.org'}],
                'exclude': ['app.models.ExcludedModel'],
            },
        ])
        self.add_group("Group title", [...])
```

But IMO it can be a lot of work to make it properly (and a good documentation is also needed).
I think you'd better implement a MixedList for now and we could open a new issue and launch a discussion on a possible new API, what do you think ?

**Comment by andybak** [comment]
Hi,

Haven't forgotten about this. My requirement for MixedList faded away but I need to do some more finessing of my dashboard UX so it will definitely come back.

Happy to discuss a new API.
